File: 020 - Factorial digit sum/Python/pe_020.py
#!/usr/bin/python
'''
file:   pe_020.py
title:  Factorial digit sum
author: Alexander P. Gosselin
e-mail: alexandergosselin@gmail.com
date:   October 8, 2014
'''

N = 100
digit_sum = 0
factorial = 1
for i in range(2, N + 1):
    factorial *= i
for c in str(factorial):
  digit_sum += int(c)

print digit_sum

File: 020 - Factorial digit sum/Haskell/pe_020.hs
{-
file:   pe_020.hs
title:  Number letter counts
author: Alexander P. Gosselin
e-mail: alexandergosselin@gmail.com
date:   May 24, 2015

link:   https://projecteuler.net/problem=20

copyright:  (C) 2015 Alexander Gosselin
license:    GNU General Public License <http://www.gnu.org/licenses/>
-}

import Data.Char (ord)

factorial :: Integer -> Integer
factorial 0 = 1
factorial 1 = 1
factorial n = n * factorial (n - 1)

pe_020 = sum $ map (flip (-) 48 . ord) $ show $ factorial 100

main :: IO ()
main = do
    print pe_020
